src/rpc/misc.cpp

```diff
@@ -1073,13 +1076,117 @@ Result
 
+    UniValue utxos = getUtxosData(vAddresses, 0, "", false, false);
+
+    if (!includeChainInfo)
+        return utxos;
+
+    UniValue result(UniValue::VOBJ);
+    result.pushKV("utxos", utxos);
+
+    {
+        LOCK(cs_main);  // for chainActive
+        result.pushKV("hash", chainActive.Tip()->GetBlockHash().GetHex());
+        result.pushKV(RPC_KEY_HEIGHT, gl_nChainHeight.load());
+    }
+    return result;
+}
+
+// insightexplorer
+UniValue getaddressutxosextra(const UniValue& params, bool fHelp)
+{
+    string disabledMsg = rpcDisabledInsightExplorerHelpMsg(RPC_API_GETADDRESSUTXOSEXTRA);
+
+    if (fHelp || params.size() != 1)
+        throw runtime_error(
+                R"(getaddressutxos {"addresses": ["taddr", ...], ("simple": true|false), ("minHeight": n)}
+
+Returns all unspent outputs for an address including inputs for the transaction (vin).
+)" + disabledMsg + R"(
+Arguments:
+{
+  "addresses":
+    [
+      "address"  (string) The base58check encoded address
+      ,...
+    ],
+  "simple"  (boolean, optional, default=false) Do not include full info about inputs with results
+  "minHeight"  (number, optional, default=0) The minimum block height to include
+  "sender"  (string, optional, default='') Filter output by sender address
+}
+
+Result
+[
+  {
+    "address"  (string) The address base58check encoded
+    "txid"  (string) The output txid
+    "height"  (number) The block height
+    "outputIndex"  (number) The output index
+    "script"  (string) The script hex encoded
+    "patoshis"  (number) The number of )" + MINOR_CURRENCY_UNIT + R"( of the output
+    "senders"  (array, optional) The inputs for the transaction
+  }, ...
+]
+
+Where "senders" is an array of objects with the following fields:
+[
+    {
+      "txid"  (string) The input txid
+      "outputIndex"  (number) The input output index
+      "address"  (string) The base58check encoded address
+      "patoshis"  (number) The number of )" + MINOR_CURRENCY_UNIT + R"( of the input
+    }, ...
+]
+OR, if input is coinbase:
+[
+    "coinbase"  (string) The coinbase hex encoded
+    ...
+]
+OR, if "simple" is true:
+[
+    "address"  (string) The base58check encoded address
+    ...
+]
+
+Examples:
+)" + HelpExampleCli(RPC_API_GETADDRESSUTXOSEXTRA, R"('{"addresses": ["tmYXBYJj1K7vhejSec5osXK2QsGa5MTisUQ"], "simple": true, "minHeight": 1000}')") +
+                HelpExampleRpc(RPC_API_GETADDRESSUTXOSEXTRA, R"({"addresses": ["tmYXBYJj1K7vhejSec5osXK2QsGa5MTisUQ"]})")
+        );
+
+    rpcDisabledThrowMsg(fInsightExplorer, RPC_API_GETADDRESSUTXOS);
+
+    bool simpleInfo = false;
+    int height = 0;
+    string senderFilter = "";
+    if (params[0].isObject())
+    {
+        UniValue simple = find_value(params[0].get_obj(), "simple");
+        if (!simple.isNull())
+            simpleInfo = simple.get_bool();
+        UniValue minHeight = find_value(params[0].get_obj(), "minHeight");
+        if (!minHeight.isNull())
```

Comment:
for minHeight param can use also get_long_number / rpc_check_unsigned_param (like in getblockmininginfo)
`